datastore/datastore.go

```diff
@@ -127,7 +129,16 @@ func (ds *Datastore) GetGetPayloadResponse(slot uint64, proposerPubkey, blockHas
 		return resp, nil
 	}
 
-	// 2. try to get from database
+	// 2. try to get from Memcached
+	resp, err = ds.memcached.GetExecutionPayload(slot, _proposerPubkey, _blockHash)
```

Comment:
can `ds.memcached` not be nil?

Reply:
it can, but since the pointer receiver returns early when memcached is nil, we can invoke this method without error on nil memcached

ideally, I think it'd be good to move towards pattern of using interfaces, such that the caller invokes interface methods instead of pointer receiever methods. That way we can initialize noop types when memcached isn't specified without sprinkling the codebase with conditional checks on struct members

Reply:
this has been updated, see: [link](https://github.com/flashbots/mev-boost-relay/pull/303#discussion_r1130896983) for discussion